File: Программирование/Лабораторные работы/Lab_2/Код/lab2_1.pas
program lab2_1;
var
 a:array[1..100] of integer;
 i, min, max: byte;
 sum, N: integer;
begin
 writeln('Введите кол-во элементов массива');
 readln(N);
 while (N < 1) or (N > 100) do begin
 writeln('Введите целое число от 1 до 100');
 readln(N);
 end;
 for i := 1 to N do begin
   writeln('Введите ',i,'-ый элемент массива');
   read(a[i]);
 end;
 writeln;
 min := 1;
 max := 1;
 for i := 2 to N do begin
   if a[i] <= a[min] then min := i else
   if a[i] >= a[max] then max := i;
 end;
 writeln(a[max],' ', a[min]);

 if min > max then begin
   i := min;
   min := max;
   max := i;
 end;

 sum := 0;
 for i := min + 1 to max - 1 do begin
   sum := sum + a[i]
 end;
 writeln(sum);
 readln;
end.

File: Программирование/Лабораторные работы/Lab_2/Код/lab2_1.cs
using System;

namespace lab2
{
    class Program
    {
        static void Main(string[] args)
        {
            int N=0;
            int[] a = new int[100];
            do
            {
                Console.Clear();
                Console.WriteLine("Введите кол-во элементов массива");
               N = Convert.ToInt32(Console.ReadLine());
            } while ((N < 1) | (N > 100));
            Console.Clear();
            Console.WriteLine("Число элементов в массиве " + N);
            for (int i = 0; i < N; i++)
            {
                Console.WriteLine("Введите " + (i+1) + "-ый элемент массива");
                a[i] = Convert.ToInt32(Console.ReadLine());
            }
            Console.Clear();
            Console.WriteLine("Число элементов в массиве " + N);
            Console.WriteLine();
            Console.Write("Массив [ ");
            for (int i = 0; i < N; i++)
            {
                Console.Write(a[i] + " ");
            }
            Console.Write("]");
            int min = 0;
            int max = 0;
            for (int i = 1; i < N; i++)
            {
                if (a[i] <= a[min])
                {
                    min = i;
                }
                else
                {
                    if (a[i] >= a[max])
                        {
                        max = i;
                    }
                }
            }
            Console.WriteLine();
            Console.WriteLine("Макс. элемент массива = " + a[max] + "         " + "Мин. элемент массива = " + a[min]);
            if (min > max)
            {
                int i;
                i = min;
                min = max;
                max = min;
            }
            int sum = 0;
            for (int i = (min + 1); i < (max); i++)
            {
                sum = sum + a[i];
            }
            Console.WriteLine("Сумма = " + sum);
        }
    }
}
